docs/sql-ref-functions-udf-scalar.md

```diff
@@ -1,22 +1,181 @@
+  </dd>
+</dl>
+
+<dl>
+  <dt><code><em>asNondeterministic(): UserDefinedFunction</em></code></dt>
+  <dd>
+    Updates UserDefinedFunction to nondeterministic.
+  </dd>
+</dl>
+
+<dl>
+  <dt><code><em>deterministic: Boolean</em></code></dt>
+  <dd>
+    Returns true iff the UDF is deterministic, i.e. the UDF produces the same output given the same input.
+  </dd>
+</dl>
+
+<dl>
+  <dt><code><em>nullable: Boolean</em></code></dt>
+  <dd>
+    Returns true when the UDF can return a nullable value.
+  </dd>
+</dl>
+
+<dl>
+  <dt><code><em>withName(name: String): UserDefinedFunction</em></code></dt>
+  <dd>
+    Updates UserDefinedFunction with a given name.
+  </dd>
+</dl>
+
+### UDFRegistration
+
+Functions for registering user-defined functions. Use `SparkSession.udf` to access this: `spark.udf`
+
+<dl>
+  <dt><code><em>register(name: String, udf: UserDefinedFunction): UserDefinedFunction</em></code></dt>
+  <dd>
+    Registers a user-defined function (UDF).
+  </dd>
+</dl>
+
+### Examples
+
+{% highlight sql %}
+
+// Define and register a zero-argument non-deterministic UDF
+// UDF is deterministic by default, i.e. produces the same result for the same input.
+// Scala
+import org.apache.spark.sql.functions.udf
+
+val foo = udf(() => Math.random())
+spark.udf.register("random", foo.asNondeterministic())
+
+sql("SELECT random()").show()
+
++------------------+
+|UDF()             |
++------------------+
+|0.9199799737037972|
++------------------+
+
+// Define and register a one-argument UDF
+// Scala
+import org.apache.spark.sql.functions.udf
+
+val plusOne = udf((x: Int) => x + 1)
+spark.udf.register("plusOne", plusOne)
+
+sql("SELECT plusOne(5)").show()
+
++------+
+|UDF(5)|
++------+
+|     6|
++------+
+
+// Define a two-argument UDF and register it with Spark in one step
+// Scala
+import import org.apache.spark.sql.functions.udf
+
+spark.udf.register("strLenScala", (_: String).length + (_: Int))
+
+sql("SELECT strLenScala('test', 1))").show()
+
++--------------------+
+|strLenScala(test, 1)|
++--------------------+
+|                   5|
++--------------------+
+
+// UDF in a WHERE clause
+// Scala
+import org.apache.spark.sql.functions.udf
+
+spark.udf.register("oneArgFilter", (n: Int) => { n > 5 })
+spark.range(1, 10).createOrReplaceTempView("test")
+
+sql("SELECT * FROM test WHERE oneArgFilter(id)").show()
+
++---+
+|id |
++---+
+|6  |
+|7  |
+|8  |
+|9  |
++---+
+
+// UDF in a GROUP BY clause
+// Scala
+import org.apache.spark.sql.functions.udf
+
+spark.udf.register("groupFunction", (n: Int) => { n > 10 })
+
+val df = Seq(("red", 1),
+             ("red", 2),
+             ("blue", 10),
+             ("green", 100),
+             ("green", 200))
+             .toDF("color", "value")
+df.createOrReplaceTempView("groupData")
+
+sql("SELECT SUM(value) FROM groupData GROUP BY groupFunction(value)").show()
+
++----------+
+|sum(value)|
++----------+
+|13        |
+|300       |
++----------+
+
+# Define and register a UDF using Python
```

Comment:
Do we need this section? I think there is doc for Python UDFs. No?

Reply:
Just want to give an example using Python. There is API doc for Python UDFs, but I won't have a separate doc for Python. I am OK to keep or delete it.